### PR TITLE
><

### PR DESCRIPTION
**Pull request by un-install**

### CODE REVIEW

File: src/main/java/lesson17/Main.java [hunk 1]
```diff
@@ -7,7 +7,18 @@
     public static void main(String[] args) throws SQLException {
```
Comment (MateAcademy) [inline src/main/java/lesson17/Main.java]: Why you wrout method main? Where are tests ????????

File: src/main/java/lesson17/OJDBCUtils.java [hunk 2]
```diff
@@ -17,4 +20,38 @@ public static Connection getConnection() {
...
+
+    public static Products newProductByRs(ResultSet rs) throws SQLException {
+        return new Products(rs.getString("MFR_ID"), rs.getString("PRODUCT_ID"),
+                rs.getString("DESCRIPTION"), rs.getBigDecimal("PRICE"), rs.getBigDecimal("QTY_ON_HAND"));
+    }
+    public static Offices newOfficeByRs(ResultSet rs) throws SQLException {
+        return new Offices(rs.getBigDecimal("office"), rs.getString("city"),
+                rs.getString("region"), rs.getBigDecimal("mgr"),
+                rs.getBigDecimal("target"), rs.getBigDecimal("sales"));
+    }
```
Comment (MateAcademy) [inline src/main/java/lesson17/OJDBCUtils.java]: It may be better to put these methods into a another class, and not where the connection to the database.

Comment (un-install) [reply src/main/java/lesson17/OJDBCUtils.java]: It is in Utils class. Why it sholdnt by in the same class with connection?

File: src/main/java/lesson17/Main.java (hunk 1 above)
Comment (un-install) [reply src/main/java/lesson17/Main.java]: Sorry. Alredy add it in 2ond commit

File: src/test/java/lesson17/OrderDAOTest.java [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+    public void testGetAllOrders2() throws SQLException {
+        Set<Orders> actual = dao.getAllOrders2();
+        assertNotNull(actual);
+    }
+
+    @Test
+    public void testFindOrderById() throws SQLException {
+        Orders actual = dao.findOrderById(id);
+        assertNotNull(actual);
+    }
```
Comment (MateAcademy) [inline src/test/java/lesson17/OrderDAOTest.java]: Wy all tests repeat at Nut null?  you can use another opportunity

File: src/main/java/lesson17/OJDBCUtils.java (hunk 2 above)
Comment (MateAcademy) [reply src/main/java/lesson17/OJDBCUtils.java]: It's my opinian, may be  SergeyPasko tell us about this situation.

File: src/main/java/lesson17/OJDBCUtils.java [hunk 4]
```diff
@@ -17,4 +20,38 @@ public static Connection getConnection() {
...
     }
+
+    public static ResultSet getResultSet(String query) throws SQLException {
+        Connection connection = getConnection();
+        PreparedStatement stmt = connection.prepareStatement(query);
+        return stmt.executeQuery();
+    }
+
+
+    public static Orders newOrderByRs(ResultSet rs, Products product) throws SQLException {
```
Comment (SergeyPasko) [inline src/main/java/lesson17/OJDBCUtils.java]: bad methods names

File: src/main/java/lesson17/OrderDAOImpl.java [hunk 5]
```diff
@@ -14,14 +14,42 @@
...
+        Set<Orders> orders = new HashSet<>();
+        while (rs.next()) {
+            orders.add(OJDBCUtils.newOrderByRs(rs, OJDBCUtils.newProductByRs(rs)));
+        }
+        OJDBCUtils.closeAllCloseble(rs, stmt, conn);
+        return orders;
     }
 
     @Override
     public Set<Orders> getAllOrders2() throws SQLException {
```
Comment (SergeyPasko) [inline src/main/java/lesson17/OrderDAOImpl.java]: bad methods names

File: src/test/java/lesson17/OrderDAOTest.java [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+import static org.junit.Assert.assertNotNull;
+
+public class OrderDAOTest {
+    private OrderDAOImpl dao = new OrderDAOImpl();
+    private static final BigDecimal id = new BigDecimal(113042);
+
+    @Test
+    public void testGetAllOrders() throws SQLException {
+        Set<Orders> actual = dao.getAllOrders();
+        assertNotNull(actual);
```
Comment (SergeyPasko) [inline src/test/java/lesson17/OrderDAOTest.java]: Your tests don't cheak any. Lock at you code. If you don't have result you return new HashSet. It empty, but not null

**Pull request merged**